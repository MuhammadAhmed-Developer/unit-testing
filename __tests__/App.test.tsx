import Home from "@/app/page"
import { screen, render } from "@testing-library/react"
test("home page", () => {
    render(<Home />)
    expect(screen.getByText("Home component")).toBeInTheDocument()
})


// Before and after hooks in jest
// beforeAll and beforeEach
//  afterAll and afterEach

beforeAll(() => {
    console.log("Before all tests")
})

afterEach(() => {
    console.log("Before each test")
})


// what is snapshot testing
// when this is usefull
// how to update snapshots

test("snapshots for home screen", () => {
    const container = render(<Home />)
    expect(container).toMatchSnapshot()
})