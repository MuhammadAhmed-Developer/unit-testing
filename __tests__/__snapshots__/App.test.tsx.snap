// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots for home screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="flex min-h-screen flex-col items-center justify-between p-24"
      >
        <div>
          Home component
        </div>
        This is main screen
        <input
          name="name"
          placeholder="input testing"
          readonly=""
          type="text"
          value="Ahmed"
        />
        <div>
          <h1>
            OnChange Event Testing
          </h1>
          <input
            type="text"
            value=""
          />
          <div>
            <h1>
              Test Click Event with Button
            </h1>
            <button>
              Update   
            </button>
            <h1 />
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="flex min-h-screen flex-col items-center justify-between p-24"
    >
      <div>
        Home component
      </div>
      This is main screen
      <input
        name="name"
        placeholder="input testing"
        readonly=""
        type="text"
        value="Ahmed"
      />
      <div>
        <h1>
          OnChange Event Testing
        </h1>
        <input
          type="text"
          value=""
        />
        <div>
          <h1>
            Test Click Event with Button
          </h1>
          <button>
            Update   
          </button>
          <h1 />
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
